src/components/lib/protdom/protdom.h

```diff
@@ -0,0 +1,60 @@
+#ifndef PROTDOM_H
+#define PROTDOM_H
+
+#include <cos_types.h>
+
+
+/* we need to expose these definitions for the static slab allocators in the booter */
+#if defined(__x86_64__)
+	#include "struct_defs_arch/x86_64/protdom_stuct_def.h"
+#else
+	#include "struct_defs_arch/default/protdom_stuct_def.h"
+#endif
+
+/* structures are architecure defined, leave them opaque to the interface */
+typedef struct protdom_ns_vas  *protdom_ns_vas_t;
```

Comment:
The structs are defined for each architecture, so why not use them here, instead of the typedef. This typedef is not like the other in that it greatly impacts how one uses the enclosed value. Hurts readability, in general to have a typedef to a pointer.

Reply:
Again, could be fine, but I want to make sure that all the trade-offs are discussed here. Generally this is a "no no".